ProjFS.Mac/PrjFSKext/VirtualizationRoots.cpp

```diff
@@ -653,3 +667,112 @@ VirtualizationRootHandle ActiveProvider_FindForPath(const char* _Nonnull path)
     
     return matchingHandle;
 }
+
+bool VirtualizationRoots_ProcessMayAccessOfflineRoots(pid_t pid)
```

Comment:
I want to make sure I understand this flow.  Is the check here to verify the request started when we were in offline mode?  How does a pid get into this array?  What are we trying to prevent here?

Reply:
This is the check for whether this is a process that has registered for requiring access to offline roots. `s_offlineIOPIDs` holds the PIDs for all processes that have opted-in to this.

Reply:
Got it so basically any pids that have triggered "PrjFS_RegisterForOfflineIO" would be allowed?

Reply:
Correct, that's the idea.